src/ws-dropdown/dropdown-menu-item.js

```diff
@@ -96,40 +108,44 @@ export class DropdownMenuItem extends Component {
    */
   onClick = event => {
     event.stopPropagation();
+    // We prevented blur with mouse down event to ensure the click is captured before the dropdown closed
+    if ('activeElement' in document) {
+      document.activeElement.blur();
+    }
+    const {item} = this.state;
     // Do nothing if this item is disabled
-    if (this.state.disabled) {
+    if (item.disabled) {
       return;
     }
     // Click on parent means back navigation
     if (this.props.isParent) {
       // dropdown-item(go-back) -> dropdown-menu(show-parent) -> dropdown-item(show-parent) -> dropdown-menu
       this.props.handle('go-back');
     // Show next menu if item has children
-    } else if (this.state.children && this.state.children.length) {
+    } else if (item.children && item.children.length) {
       this.props.handle('show-child', this.menu);
-    } else {
-      if (!this.context.multiple) {
-        // If it is selected we publish null because it will be deselected in the upper menu
-        if (this.state.selected) {
-          this.props.handle('change', null);
-        } else {
-          this.setState({
-            selected : true,
-            stored : true,
-          });
-
-          this.props.handle('change', this.state);
-        }
+    } else if (!this.context.multiple) {
+      // If it is selected we publish null because it will be deselected in the upper menu
+      if (item.selected) {
+        this.props.handle('change', null);
       } else {
-        this.setState({selected : !this.state.selected});
+        item.selected = true;
+        item.stored = true;
+
+        this.setState({item});
+
+        this.props.handle('change', item);
       }
+    } else {
+      item.selected = !item.selected;
+      this.setState({item});
```

Comment:
same here I would say.

Reply:
It's true that this.state probably is already modified but I don't see a problem here as the component is rendered only when calling setState. Or did I miss something?